File: PSET1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float price;
    do
    {
        price = get_float("Change owed: ");
    }
    while (price < 0);

    int change = round(price * 100);
    printf("%i\n\n", change);

    int q = change / 25;
    printf("Number of quarters: %i\n", q);

    int d = (change - (q * 25)) / 10;
    printf("Number of dimes: %i\n", d);

    int n = (change - q * 25 - d * 10) / 5;
    printf("Number of nickels: %i\n", n);

    int p = (change - q * 25 - d * 10 - n * 5);
    printf("Number of pennys: %i\n\n", p);

    int t = q + d + n + p;
    printf("%i\n", t);
}

File: PSET6/cash.py
from cs50 import get_float


def main():
    while True:
        change = get_float("Change owed: ")
        if change > 0:
            break

    cents = int(change * 100)

    print("Your change is:", cents)

    # check for quarters
    q = int(quarters(cents))
    print("Quarters:", q)
    
    # check for dimes
    d = int(dimes(cents, q))
    print("Dimes:", d)

    # check for nickels
    n = int(nickels(cents, q, d))
    print("Nickels:", n)

    # check for pennies
    p = int(pennies(cents, q, d, n))
    print("Pennies:", p)

    # add coins & print
    sum = q + d + n + p
    print(f"{sum}")


###############################
def quarters(cents):
    q = int(cents / 25)
    return q


def dimes(cents, q):
    d = int((cents - q*25) / 10)
    return d


def nickels(cents, q, d):
    n = int((cents - q*25 - d*10) / 5)
    return n


def pennies(cents, q, d, n):
    p = ((cents - q*25 - d*10 - n*5) / 1)
    return p
    
    
main()